### DISCUSSION TITLE
DataFrame.to_csv()

### DISCUSSION THREAD

**Discussion by yohanespm77** [teamalgoritma/community#304] [Q&A] [answered]
Saya melakukan perhitungan rata-rata pada data saya seperti berikut :
<img width="373" alt="image" src="https://user-images.githubusercontent.com/106010284/189859625-5ebc9cf4-1bb6-4a79-b683-149d32854251.png">

<img width="697" alt="image" src="https://user-images.githubusercontent.com/106010284/189858449-d5a8c42b-e404-4880-bbde-fe27cc62eaf3.png">

Ketika saya simpan dalam bentuk csv, hasilnya ada yang berubah menjadi format scientific
<img width="676" alt="image" src="https://user-images.githubusercontent.com/106010284/189859028-17490bf5-13be-421f-99b3-b2dbf50a26f7.png">

Solusinya gimana ya supaya tidak berubah di csvnya ?

**Comment by taruma** [comment, marked as answer]
bisa menambahkan argumen `float_format=` saat menggunakan _method_ `.to_csv(...)`

```python
>>> df[['avg']].to_csv("latihan.csv", float_format="{:.2f}".format)
```

untuk formatnya digunakan `.2f`, biar nanti program lain bisa langsung membaca jenis datanya (_decimal_/_float_). Kalau ditambah _separator_ `,`, nanti perlu proses pengolahan lebih lanjut di program lain atau pembacaan ulang di pandas. 

referensi lanjut:
- https://pyformat.info/
- https://stackoverflow.com/questions/12877189/float64-with-pandas-to-csv
